kv: return the actual epoch in replicaRLockedStoreLiveness.SupportFrom

nvb opened the pull request:
This is needed to complete the intention of https://github.com/cockroachdb/cockroach/pull/131614.

May explain recent `epochs in store liveness shouldn't regress` assertion failures.

Epic: None
Release note: None

nvb committed:
kv: return the actual epoch in replicaRLockedStoreLiveness.SupportFrom

This is needed to complete the intention of #131614.

Epic: None
Release note: None

nvb committed:
kv: log on missing store in replicaRLockedStoreLiveness

Assists with debugging.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/133199)
<!-- Reviewable:end -->

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/133199#-:-O9qIUoe-gz7BchOVV1V:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @miraradeva and @nvanbenschoten)

___
*[`pkg/kv/kvserver/replica_store_liveness.go` line 62 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/133199#-O9qII1x725laSGoUAlG:-O9qII1z5Ej-RDqAuTZB:bwq83rg) ([raw file](https://github.com/cockroachdb/cockroach/blob/95a020d6d121a0c8b0502816962062f4ede10b91/pkg/kv/kvserver/replica_store_liveness.go#L62)):*
> ```Go
> 	storeID, ok := r.getStoreIdent(replicaID)
> 	if !ok {
> 		return 0, false
> ```

Should we add a log line for when we're in this case as well, while we're here?

EDIT: ofc it's in the second commit :joy: 
___
*[`pkg/kv/kvserver/replica_store_liveness.go` line 64 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/133199#-O9qIDwTE5BUz80jjrTf:-O9qIDwTE5BUz80jjrTg:beol453) ([raw file](https://github.com/cockroachdb/cockroach/blob/95a020d6d121a0c8b0502816962062f4ede10b91/pkg/kv/kvserver/replica_store_liveness.go#L64)):*
> ```Go
> 		return 0, false
> 	}
> 	// TODO(arul): we can remove this once we start to assign storeLiveness in the
> ```

Wanna address this TODO as well, while you're here?


<!-- Sent from Reviewable.io -->

nvb committed:
kv: address TODO in replicaRLockedStoreLiveness.SupportFor

We now assign storeLiveness in the Store constructor.

Epic: None
Release note: None

nvb reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/133199#-:-O9qRnLB7mLmA-bDTHjO:b-oduykh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani and @miraradeva)

___
*[`pkg/kv/kvserver/replica_store_liveness.go` line 64 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/133199#-O9qIDwTE5BUz80jjrTf:-O9qRq1f1UMTKVKqpvqK:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/95a020d6d121a0c8b0502816962062f4ede10b91/pkg/kv/kvserver/replica_store_liveness.go#L64)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Wanna address this TODO as well, while you're here?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
